### DISCUSSION TITLE
What is the general procedure for omitting tools/framework from the image creation?

### DISCUSSION THREAD

**Discussion by MichaelRandall-CFS** [actions/virtual-environments#7757] [General questions]
What is the general procedure for omitting tools from the build?

for instance, if you don't want java, rust, go, ruby

I have tried a few things; however they seem to create errors in the build.. 

I was wondering if there is just something simple, I can do to speed up the creation time by omitting tools and frameworks i don't need. If there is documentation on this, or a previous questions id be more than happy for the details

**Comment by ilia-shipitsin** [comment]
if you mean a posibility to add or not to add tools, I'm afraid it will add complexity, we have at least 3 stages

1) create image
2) test image
3) create software report

all stages should work with either combination of tools installed.
it looks complicated, but if you are interested, you can create PR

**Comment by vpolikarpov-akvelon** [comment]
Hi, @RbsTechOnline. There is no common procedure for enabling/disabling tools in image unfortunately. As @ilia-shipitsin already mentioned it would add complexity that we have enough already.

Anyway, this repo contains scripts for building images that is used to provision GitHub-hosted runners. These runners should be as versatile as possible so images contain tons of different software. I'm sure you don't need most of the tools and framework we are installing here and I would suggest to start from minimal working image and just add only what you really need. You can use [this template](https://github.com/actions/runner-images/blob/main/images/linux/ubuntuminimal.pkr.hcl) for example, it builds in bare 10 - 15 minutes.
